msi/GitProduct.wxs

```diff
@@ -28,13 +28,99 @@
     <!-- Other MSI specific properties -->
     <Property Id='MSIFASTINSTALL' Value='7' />
 
+    <!-- Git for Windows MSI Properties -->
+    <!-- CLIENVIRONMENT Property implementing the "Remember Property" pattern -->
+    <Property Id="CLIENVIRONMENT" Value="Bash">
+      <RegistrySearch Id="CliEnvironmentSearch" Root="HKLM" Key="SOFTWARE\GitForWindows" Name="CliEnvironment" Type="raw" />
+    </Property>
+    <SetProperty Id="CMDLINE_CLIENVIRONMENT" Value="[CLIENVIRONMENT]" Before="AppSearch" Sequence="first" />
+    <SetProperty Id="CLIENVIRONMENT" Value="[CMDLINE_CLIENVIRONMENT]" After="AppSearch" Sequence="first">CMDLINE_CLIENVIRONMENT</SetProperty>
```

Comment:
After working on this over and over again, I completely agree -- it's best to create these properties when I implement configuring the option in question. I can't introduce the UI piecemeal because of the interdependencies, but the properties I can and it makes sense at this point.